.github/workflows/docker-checks.yml

```
@@ -0,0 +1,33 @@
+# Copyright (c) 2024 Hemi Labs, Inc.
+# Use of this source code is governed by the MIT License,
+# which can be found in the LICENSE file.
+
+name: Docker Checks
+
+on:
+  workflow_call:
+    inputs:
+      context:
+        description: "The build context"
+        default: "."
+        type: string
+      fetch-depth:
+        description: Number of commits to fetch. 0 indicates all history for all branches and tags.
+        default: 1
+        type: number
+
+jobs:
+  docker-checks:
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: docker build --tag ${{ github.repository }}:${{ github.sha }} ${{ inputs.context }}
+      - uses: aquasecurity/trivy-action@0.29.0
```

As this is a third-party action, not maintained by either GitHub or Docker, we should pin it to a commit hash  in order to make it immutable.

```suggestion
      - uses: aquasecurity/trivy-action@76071ef0d7ec797419534a183b498b4d6366cf37 # 0.29.0
```

https://github.com/aquasecurity/trivy-action/commit/76071ef0d7ec797419534a183b498b4d6366cf37

In addition to being third-party, this is also a security scanning tool.